src/Disks/ObjectStorages/ObjectStorageFactory.cpp

```diff
@@ -73,9 +73,17 @@ ObjectStoragePtr createObjectStorage(
         return std::make_shared<PlainObjectStorage<BaseObjectStorage>>(std::forward<Args>(args)...);
     else if (isPlainRewritableStorage(type, config, config_prefix))
     {
-        /// TODO(jkartseva@): Test support for generic disk type
-        if (type != ObjectStorageType::S3)
-            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "plain_rewritable metadata storage support is implemented only for S3");
+        /// HDFS object storage currently does not support iteration and does not implement listObjects method.
```

Comment:
JFYI, listObjects for hdfs object storage is added in https://github.com/ClickHouse/ClickHouse/pull/59767, so this can be adjusted afterwards.